Extensibility sources, samples and documentation (#284)
Starcounter/Docs: extensibility/README.md

# Overview
Extensibility is about extending core features of the Starcounter database, using some suggested approaches available here as source code.

> ## Organisation

Reviewer: ```suggestion
## File structure
```

The [Starcounter.Database.Hosting.Extensibility](./Starcounter.Database.Hosting.Extensibility) folder contains sources that forms the foundation for the more specific samples.

### OnDelete

> * [OnDelete](./OnDelete) suggest a way for any application to support an interface that can be applied to any database class and act as a callback when instances of that class is deleted from the database.

Reviewer: ```suggestion
* [OnDelete](./OnDelete) demonstrates a way for any application to support an interface that can be applied to any database class and act as a callback when instances of that class is being deleted from the database.
```

### Pre-commit hooks

> * [PreCommitHooks](./PreCommitHooks) describe an approach where applications can register callbacks to be invoked when instances of a specified database type is either inserted or updated, just before a transaction is committed. 

Reviewer: ```suggestion
* [PreCommitHooks](./PreCommitHooks) demonstrates an approach where applications can register callbacks to be invoked when instances of a specified database type is either inserted or updated, just before a transaction is committed. 
```
